process/src/services/service.ports.ts

```diff
@@ -241,11 +242,13 @@ class ServicePorts {
     private _updateConnectedPortsState() {
         clearTimeout(this._connectedPortsState.timer);
         this._connectedPortsState.state = {};
+
         this._controllers.forEach((controller: ControllerSerialPort) => {
             this._connectedPortsState.state[controller.getPath()] = {
                 connections: this._getConnectionsCount(controller.getPath()),
                 ioState: controller.getIOState()
             };
+            controller.clearIOState();
```

Comment:
Each controller is created for one port. Probably we can store a state of port just inside the controller and get rid of common storage in service.ports?

Reply:
I'm not sure what exactly you mean since the state is inside the controller, there is no common storage in service.ports

Reply:
we can leave it for refactoring stafe